src/cpp/rtps/transport/asio_helpers.hpp

```diff
@@ -87,11 +87,11 @@ struct asio_helpers
             // Last attempt was successful. Get the actual value set.
             BufferOptionType option;
             socket.get_option(option, ec);
-            if (!ec)
+            if (!ec && (option.value() >= value_to_set))
```

Comment:
Super NIT: I wonder if it's possible that set results in a value greater than the one specified. I guess not, but if that was the case maybe we should fail? Or add an extra check that the value isn't greater than `initial_buffer_value`.

Reply:
2eec3ba